crypto/x509: system root pool is not available on Windows causes TestClientTLS to fail

As go-lang does not implement system CertPool on windows, the TLS test in main_test.go fails : 
```console
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x20 pc=0x610de2]

goroutine 1 [running]:
crypto/x509.(*CertPool).AddCert(0x0, 0xc0420c9600)
        C:/Go/src/crypto/x509/cert_pool.go:95 +0x72
crypto/x509.(*CertPool).AppendCertsFromPEM(0x0, 0xc042182000, 0x442, 0x642, 0x642)
        C:/Go/src/crypto/x509/cert_pool.go:128 +0x144
github.com/dedis/drand/net.(*CertManager).Add(0xc0420880c8, 0xc042090040, 0x32, 0xc042090040, 0x32)
        C:/Users/mathi/go/src/github.com/dedis/drand/net/certs.go:37 +0x75
main.fetchPrivateCmd(0xc0420ab340, 0xc042083f20, 0x8a2208)
        C:/Users/mathi/go/src/github.com/dedis/drand/main.go:357 +0x4cd
main.main.func7(0xc0420ab340, 0x0, 0xc0420ab340)
        C:/Users/mathi/go/src/github.com/dedis/drand/main.go:174 +0x32
github.com/dedis/drand/vendor/github.com/urfave/cli.HandleAction(0x91ba40, 0xa1b470, 0xc0420ab340, 0xc042124500, 0x0)
        C:/Users/mathi/go/src/github.com/dedis/drand/vendor/github.com/urfave/cli/app.go:490 +0xcf
github.com/dedis/drand/vendor/github.com/urfave/cli.Command.Run(0x9f5286, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0xa130f3, 0x4c, 0x0, ...)
        C:/Users/mathi/go/src/github.com/dedis/drand/vendor/github.com/urfave/cli/command.go:210 +0xa3d
github.com/dedis/drand/vendor/github.com/urfave/cli.(*App).RunAsSubcommand(0xc0420a3040, 0xc0420ab080, 0x0, 0x0)
        C:/Users/mathi/go/src/github.com/dedis/drand/vendor/github.com/urfave/cli/app.go:379 +0xa17
github.com/dedis/drand/vendor/github.com/urfave/cli.Command.startApp(0x9f31e3, 0x5, 0x0, 0x0, 0xc042055380, 0x1, 0x1, 0x9fda25, 0x15, 0x0, ...)
        C:/Users/mathi/go/src/github.com/dedis/drand/vendor/github.com/urfave/cli/command.go:298 +0x8e0
github.com/dedis/drand/vendor/github.com/urfave/cli.Command.Run(0x9f31e3, 0x5, 0x0, 0x0, 0xc042055380, 0x1, 0x1, 0x9fda25, 0x15, 0x0, ...)
        C:/Users/mathi/go/src/github.com/dedis/drand/vendor/github.com/urfave/cli/command.go:98 +0x124d
github.com/dedis/drand/vendor/github.com/urfave/cli.(*App).Run(0xc0420a2d00, 0xc042092000, 0x6, 0x8, 0x0, 0x0)
        C:/Users/mathi/go/src/github.com/dedis/drand/vendor/github.com/urfave/cli/app.go:255 +0x6a7
main.main()
        C:/Users/mathi/go/src/github.com/dedis/drand/main.go:187 +0x1af6

--- FAIL: TestClientTLS (6.96s)
        Error Trace:    main_test.go:134
        Error:          Received unexpected error:
                        exit status 2
        Test:           TestClientTLS
FAIL
```

--- a/main.go
+++ b/main.go
@@ -10,6 +10,7 @@ import (
 	"path"
 	"path/filepath"
 	"regexp"
+	"runtime"
 	"strconv"
 	"strings"
 
@@ -191,6 +192,10 @@ func main() {
 }
 
 func keygenCmd(c *cli.Context) error {
+	//x509 not available on windows: must run in insecure mode
+	if runtime.GOOS == "windows" && !c.Bool("insecure") {
+		slog.Fatal("TLS is not available on Windows, please run in insecure mode")
+	}
 	args := c.Args()
 	if !args.Present() {
 		slog.Fatal("Missing drand address in argument (IPv4, dns)")
@@ -328,6 +333,10 @@ func beaconCmd(c *cli.Context) error {
 }
 
 func runCmd(c *cli.Context) error {
+	//x509 not available on windows: must run in insecure mode
+	if runtime.GOOS == "windows" && !c.Bool("insecure") {
+		slog.Fatal("TLS is not available on Windows, please run in insecure mode")
+	}
 	conf := contextToConfig(c)
 	fs := key.NewFileStore(conf.ConfigFolder())
 	var drand *core.Drand

